### PR TITLE
feat: Add `autoscaling_group_tags` variable to self-managed-node-groups

### LINKED ISSUE #2072: Support for autoscaling_group_tags in self-managed-node-group module
## Is your request related to a problem? Please describe.

`cluster-autoscaler` auto-discovery setup relies on tags placed in the ASG (see: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#auto-discovery-setup).

However, the variable `tags` in the `self-managed-node-group` also applies the tags to `security_groups`, `iam_roles` and `instance_profile` - which does not make sense to have the same tags.

We can't also use `aws_autoscaling_group_tag` resource because they should not be combined with managing tags via `aws_autoscaling_group` directly (https://github.com/terraform-aws-modules/terraform-aws-eks/blob/d969e94bbea89ed616713bdab849f042ceda87ac/modules/self-managed-node-group/main.tf#L386).

## Describe the solution you'd like.

A `autoscaling_group_tags` variable, similarly to `iam_role_tags`, `launch_template_tags`, and `security_group_tags`.

### PR DESCRIPTION
**Pull request by gferon**
## Description

`cluster-autoscaler` discovers ASGs using tags, which could currently only be set by setting tags on all resources. This helps setting tags like `k8s.io/cluster-autoscaler/enabled` on the `aws_autoscaling_group` resources only.

## Motivation and Context

Fixes #2072 

## Breaking Changes

None

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by bryantbiggs** [comment]
Thanks for the PR @gferon - lets add an entry in `node_groups.tf` for this below the default tags https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/node_groups.tf#L391 as well

**Comment by gferon** [comment]
> Thanks for the PR @gferon - lets add an entry in `node_groups.tf` for this below the default tags https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/node_groups.tf#L391 as well

Indeed, I forgot to push my last commit.

File: examples/self_managed_node_group/main.tf [hunk 1]
```diff
@@ -97,6 +97,12 @@ module "eks" {
 
   self_managed_node_group_defaults = {
     create_security_group = false
+
+    # enable discovery of autoscaling groups by cluster-autoscaler
+    autoscaling_group_tags = {
+      "k8s.io/cluster-autoscaler/enabled" : true,
+      "k8s.io/cluster-autoscaler/${local.name}" : true,
```
Comment (matthewdaltonfanduel) [inline examples/self_managed_node_group/main.tf:104]: per https://docs.aws.amazon.com/eks/latest/userguide/autoscaling.html should this be `owned` ?

File: modules/self-managed-node-group/variables.tf [hunk 2]
```diff
@@ -464,6 +464,12 @@ variable "use_default_tags" {
   default     = false
 }
 
+variable "autoscaling_group_tags" {
+  description = "A map of additional tags to add to the auto_scaling_group created"
```
Comment (bryantbiggs) [inline modules/self-managed-node-group/variables.tf:468]:
```suggestion
  description = "A map of additional tags to add to the autoscaling group created. Tags are applied to the autoscaling group only and are NOT propagated to instances"
```

File: modules/self-managed-node-group/README.md [hunk 3]
```diff
@@ -77,6 +77,7 @@ module "self_managed_node_group" {
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
 | <a name="input_ami_id"></a> [ami\_id](#input\_ami\_id) | The AMI from which to launch the instance | `string` | `""` | no |
+| <a name="input_autoscaling_group_tags"></a> [autoscaling\_group\_tags](#input\_autoscaling\_group\_tags) | A map of additional tags to add to the auto\_scaling\_group created | `map(string)` | `{}` | no |
```
Comment (bryantbiggs) [inline modules/self-managed-node-group/README.md:80]:
```suggestion
| <a name="input_autoscaling_group_tags"></a> [autoscaling\_group\_tags](#input\_autoscaling\_group\_tags) | A map of additional tags to add to the autoscaling group created. Tags are applied to the autoscaling group only and are NOT propagated to instances | `map(string)` | `{}` | no |
```

**Review by bryantbiggs — APPROVED** [review]
Thanks for the PR @gferon !

**Comment by antonbabenko** [comment]
This PR is included in version 18.23.0 :tada:

**Comment by kirrmann** [comment]
Any reason why this was not done for eks managed node groups?

**Pull request merged**